src/service/resources-impl.js

```diff
@@ -332,6 +332,27 @@ export class Resources {
   }
 
   /**
+   * Returns a promise to the layoutBox for the element. If the element is
+   * resource-backed then makes use of the resource layoutBox, otherwise
+   * measures the element directly.
+   * @param {!Element} element
+   * @return {!Promise<!../layout-rect.LayoutRectDef>}
+   */
+  getElementLayoutBox(element) {
```

Comment:
I'm ok with this. But /cc @jridgewell - this is a good case for the kind of data that layers system should be able to produce in the future.